js/src/main/scala/Main.scala

```diff
@@ -138,13 +104,89 @@ object Main {
     )
   }
 
+  // Extract a separated function to do this.
+  // Returns `Left[Str]` if successful, `Right[Str]` if not.
```

Comment:
Note that the convention is `Right` for successful (it also sounds better, being able to say "the _right_ result").

Reply:
Changed.